app/src/shared/components/TextField/NumberField/index.jsx

```diff
@@ -59,15 +60,24 @@ const NumberField = ({
             newValue = max
         }
 
-        setInternalValue(newValue.toString())
-    }, [min, max, internalValue])
+        if (inputRef.current) {
+            inputRef.current.focus()
+        }
+
+        // eslint-disable-next-line no-param-reassign
+        event.target.value = newValue.toString()
```

Comment:
this seems weird

Reply:
I agree, it's a hack - but I couldn't think of a better way without refactoring the whole thing and how the value gets saved since number input is baked into the `TextInput`.